File: LeetCode 30 Day Challenge/08-middleOfLinkedList.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}

var middleNode = function(head) {
    let first = head
    let second = head
   
    while (first && first.next) {
        second = second.next
        first = first.next.next
    }
    return second
};

File: LeetCode 30 Day Challenge/08-middle_of_linked_list.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def middleNode(self, head):
        first = head
        second = head
        
        while first and first.next:
            second = second.next
            first = first.next.next
            
        return second